### PR TITLE
adding assert messages to std.conv

### PR DESCRIPTION
**Pull request by burner**

### CODE REVIEW

File: std/conv.d [hunk 1]
```diff
@@ -1469,7 +1469,8 @@ if (!isImplicitlyConvertible!(S, T) &&
         }
         else
         {
-            static assert(tSmallest < 0);
+            static assert(tSmallest < 0,
+                "minimum value of T most be smaller than 0");
```
Comment (berni44) [inline std/conv.d]: most => must

Comment (burner) [reply std/conv.d]: thank you

File: std/conv.d [hunk 2]
```diff
@@ -6328,7 +6330,7 @@ if ((radix == 2 || radix == 8 || radix == 10 || radix == 16) &&
         else static if (radix == 16)
             enum SHIFT = 4;
         else
-            static assert(0);
+            static assert(false, "radix must be 2, 8, or 16");
```
Comment (berni44) [inline std/conv.d]: could also be 10...

Comment (burner) [reply std/conv.d]: not in that branch, I believe.

Comment (berni44) [reply std/conv.d]: ah ok, I see

File: std/conv.d [hunk 3]
```diff
@@ -5813,14 +5823,6 @@ if (isIntegral!T)
...
 }
 
-auto unsigned(T)(T x)
-if (isSomeChar!T)
-{
-    // All characters are unsigned
-    static assert(T.min == 0);
-    return cast(Unqual!T) x;
-}
-
```
Comment (n8sh) [inline std/conv.d]: Why is this being removed?

Comment (burner) [reply std/conv.d]: its not it was moved for the /// Ditto

File: std/conv.d (hunk 2 above)
Comment (n8sh) [inline std/conv.d]:
```suggestion
            static assert(false, "radix must be 2, 8, 10, or 16");
```

The radix 10 case is earlier in this function.

**Comment by burner** [comment]
thank you @berni44 @n8sh

**Pull request merged**